cloud/amazon/iam_saml_federation.py

```diff
@@ -0,0 +1,185 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = '''
+---
+module: iam_saml_federation
+version_added: "2.3"
+short_description: maintain iam saml federation configuration.
+description:
+    - Maintains IAM SAML identity federation configurations.  This module has a dependency on python-boto >= 2.5.
+options:
+  name:
+    description:
+      - The name of the provider to create.
+    required: true
+  saml_metadata_document:
+   description:
+      - An  XML  document  generated by an identity provider (IdP) that supports SAML 2.0.
```

Comment:
Can you link to the AWS docs that explain this XML doc? The only reference I can find is docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_providers_saml.html